src/drivers/wayland/lv_wayland.c

```diff
@@ -2221,6 +2224,11 @@ static void _lv_wayland_flush(lv_display_t * disp, const lv_area_t * area, unsig
     /* Modify specified area in buffer */
     for(y = area->y1; y <= area->y2; y++) {
         offset = ((area->x1 + (y * hres)) * bpp);
+        if(app->shm_format == WL_SHM_FORMAT_ARGB8888) {
+            for(x = area->x1; x <= area->x2; x++) {
+                lv_color_premultiply((lv_color32_t *)color_p + (x - area->x1));
+            }
```

Comment:
Have less operations in the loop
```suggestion
            int32_t w = lv_area_get_width(area);
            for(x = 0; x < w; x++) {
                lv_color_premultiply((lv_color32_t *)color_p + x);
            }
```

Reply:
I'm not sure if this will effectively compile in less operations. 

```
x = area->x1 // assign

x <= area->x2 // compare
(lv_color32_t *)color_p + (x - area->x1) // sum
```
vs

```
x = 0 // assign

x < w // compare
(lv_color32_t *)color_p + x // sum
```

The compiler is smart enough to combine `(lv_color32_t *)color_p - area->x1`, to end up in one operation.

The reason I wrote it like this, is simply to have some analogy/consistency with the vertical for loop.

Another option would have been to re-use the src_width/src_height variables. (Your lv_area_get_width in fact), but then I would have to change the vertical for loop to keep it consistent.

Reply:
Update: I reworked the for loops slightly.